### PR TITLE
GNU/kFreeBSD support

### PR DESCRIPTION
**Pull request by stevenc99**
Hi,  these commits fix issues seen on Debian GNU/kFreeBSD, as mentioned in https://github.com/mamedev/mame/pull/1093

Thanks!

### CODE REVIEW

File: include/bx/thread.h [hunk 1]
```diff
@@ -157,12 +165,10 @@ namespace bx
 		{
 #if BX_PLATFORM_OSX || BX_PLATFORM_IOS
 			pthread_setname_np(_name);
```
Comment (bkaradzic) [inline include/bx/thread.h]: Can you move just implementation of pthread_setname_np into compatibility headers?

**Comment by stevenc99** [comment]
I guess not... because the function call is different for each platform and in the case of Windows it takes extra parameters.  I'm not sure what you had in mind.

The compat headers system doesn't adapt well to the fact a platform could be based on FreeBSD *and* have glibc, or have Linux *without* glibc, and other combinations that the ifdefs already handle with better granularity.

**Comment by bkaradzic** [comment]
I'll add BX_CRT_<> macros to specify which C runtime library is used.

**Comment by bkaradzic** [comment]
Added it: https://github.com/bkaradzic/bx/blob/master/include/bx/platform.h#L32

**Pull request closed without merging**